### ISSUE TITLE
Experiments on autoencoder architecture

### ISSUE THREAD

**Issue by JonnyyTorres** [JonnyyTorres/Galsim_JAX#2]
Hello @b-remy and @EiffL.  I would like to show you some of the results of changing the encoder and decoder depth on the log likelihood obtained. In principle, I defined 3 models with the following characteristics that you can see below: 

https://github.com/JonnyyTorres/Galsim_JAX/blob/52bdb16938688207491ca1251788ad88555c91be/VAE_Resnet_train.py#L139-L167

Basically, I used the same convolution and transposed convolution operations for the encoder and decoder, varying the depth of each model to a latent space dimension of 64, 32 and 16 respectively.

https://wandb.ai/jonnyytorres/resnet-comp-dim/reports/test_log_likelihood-23-06-13-13-43-18---Vmlldzo0NjMyMTA5 

For the 3 experiments I used a total of 50k steps. I must say that the results on the log likelihood on the test data are revealing, insofar as they show that with a dimension of 64 in the latent space, the autoencoder turns out to be more appropriate to capture the details of the original data. On the other hand, with a dimension of 32, although low log likelihood values are achieved, they are not as good as in the first case. Finally, for the dimension of 16, apparently quite a lot of information is lost from the original data, being reflected in higher values of the log likelihood, with a higher variance as well, as can be perfectly appreciated in the case of this metric for the training data. 

https://wandb.ai/jonnyytorres/resnet-comp-dim/reports/log_likelihood-23-06-13-13-48-45---Vmlldzo0NjMyMTcx

With this in mind, I would like to hear your thoughts and opinions on the matter. At Benjamin's suggestion, I will focus my next analysis on changing the different operations and activation functions between layers, to see if there is an improvement on the loss obtained in the case of having a dimension of 64 in the latent space and with a regularization factor of 0.01 which has been the factor with the best results so far.

**Comment by EiffL** [comment]
This is great @JonnyyTorres ! But it looks like you are changing 2 parameters at once, the depth of the model, and the size of the latent space.

Could you show us images from the results as well, so that we can better associate visual performance with the value of the log likelihood?

Also, what is the default architectural choice used in a model like Stable Diffusion? I think we could use their architecture as a baseline, and fine tune it a bit for our purposes. But it might be a good way to do architecture tuning from a good baseline

**Comment by JonnyyTorres** [comment]
Hello @EiffL, according to your suggestion, I retrained two models varying only the dijmension of the latent space without changing the depth of the model. I share with you a report I made based on the results obtained:

https://wandb.ai/jonnyytorres/resnet-comp-dim/reports/Comparison-in-the-depth-of-latent-space--Vmlldzo0NjMzMjIz

As you can see, the results obtained are not very conclusive when comparing the log likelihood values for the 32 and 64 dimensions, however, the results for a latent dimension of 64 are still slightly better. 

As for a dimension of 16, I think we start to lose finer details of the original images, which goes against our objective, so I consider that we could call it an asymptotic dimension (if the term fits in this context).
